### ISSUE TITLE
Remove undetermined from language selection

### ISSUE THREAD

**Issue by Tealk** [LemmyNet/lemmy-ui#1166]
### Is your proposal related to a problem?

Often "undetermined" is offered and set as the default language, and I often see posts where no language is selected.
This bypasses all the language filtering.

### Describe the solution you'd like

That the selection "undetermined" is hidden when creating contributions and comments

**Comment by Nutomic** [comment]
Undetermined should be the default, we cant expect users to set the language manually all the time. However I suppose it would be fine to remove it as a dropdown option.

**Comment by Chais** [comment]
I agree it should be the overall default, but it should also be configurable. Ideally on every level.  
I imagine something like this:

- A fresh instance's language defaults to "undetermined" but the admin can configure it.
- Each community on that instance defaults to the instance's language, with an option to override (helpful for multilingual instances)
- When posting to a community the language defaults to the community's language with an option to override (multilingual instances again)
- When commenting on a post the language defaults to the post's language with an option to override (bit of an edge-case. for example English natives asking for resources for learning French in a French community on a bilingual instance)
- When replying to a comment the language defaults to the comment's language with an option to override again (I don't expect this to be useful very often, but at this point, why divert from the pattern ¯\\\_(ツ)\_/¯ )

This way we establish sane defaults on every level, but there's always the option to change it. But having sane defaults means users should rarely have to change them.

**Comment by kevincox** [comment]
Another idea:

Something that would help a lot is allowing user to select their "primary posting languages". I basically only use English online but I honestly get tired of selecting it for every comment. So I probably send a lot of "undetermined" posts and comments out of laziness. For my use case being able to set the default language would mean lots of accurately tagged posts and comments.

Obviously it is more complicated for people who speak multiple languages. But maybe if they could select their "primary posting lagnuages" and it could be intersected with the languages allowed in the communities? If there is not a unique overlap at least their primary languages could be at the top of the list to make it easier than hunting through the entire list. Even then I suspect lots of multi-lingual people have a strong bias towards a particular language that it may make sense to have it set as the default for them, however this risks mis-tagging if they forget to switch the default which is probably worse than "undetermined".

This may be useful in compliment to @Chais's idea. Although quite possibly if most instances/communities gain a language tag then personal preference will be mostly irrelevant. Although it could help a bit for multilingual communities. Although these are fairly rare they definitely exist, especially for things like cities. But in this case it may be better just to follow the parent as suggested above. Only in a situation like the top-level post would the user's language be a main default.

Pros:
- Allows user configuration. The user is very rarely going to post in languages they don't speak which tends to be a fairly short list.
- Works well even if communities aren't commonly tagged.
- Provides some disambiguation for multilingual communities.

Cons:
- Doesn't work particularly well for participating in different communities with different languages (although tagged communities will mostly fix this).
- May be mostly unnecessary in the case of tagged communities.

**Comment by Chais** [comment]
That approach only gets different results than what I described when you insist on commenting/replying in a language foreign to the community/thread.  
If you're default is English and you reply to a French thread my logic would default to French (with the option to change it) while yours would default to English.  
Is argue that sticking with the thread's language is the better default, as most users will do that.

**Comment by kevincox** [comment]
Yes, alone the behaviour isn't the same. But it works better if the thing you are replying to isn't tagged.

I think if we combined the two approaches it would use the "inheritance" approach primarily but then fall back to user preference if there is no overlap or the parent isn't tagged.

**Comment by Nutomic** [comment]
FYI the backend already chooses a default language when possible. Namely if no language is specified, and the overlap of community languages and user languages only contains a single item (excluding undefined), that is used as default. So this logic should be duplicated in lemmy-ui.

**Comment by Tealk** [comment]
> FYI the backend already chooses a default language when possible. 

Shouldn't the automatically selected language also be displayed in the dropdown? That's why many users are confused.

However, I see many posts that do not have an assigned language, so something should be done.

**Comment by Nutomic** [comment]
No, its selected after you submit the post, before its written to the database and federated.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/LemmyNet/lemmy-ui/issues/1166#issue-1751305641

@Tealk, there are many instances in which a user would not wish to choose a language. Consider the option for "English" (`eng`). That encompasses variants like `eng-GBR-oxendict`, and probably even `eng-GBR-oxendict-fonipa` (its IPA representation). However, does it encompass derivative conlangs? What about languages that are not listed in ISO 639? In that case, "Undetermined" corresponds directly to `und`, and can reasonably be considered to encompass `mul`, too. Removing this option forces me to *incorrectly* pick a language tangentially related at best for such content.

**Comment by Tealk** [comment]
> _many_ instances exist in which a user would not wish to choose a language. Consider the option for "English" (`eng`).

Why would you not want to specify a language?

**Comment by RokeJulianLockhart** [comment]
> https://github.com/LemmyNet/lemmy-ui/issues/1166#issuecomment-2600870258

@Tealk, not *not a language*, per se. That was poor phrasal. (However, the subsequent content should have inherently clarified.) Rather, ISO 639 doesn't encompass all languages. As aforestated, this includes conlangs and niche (usually local) derivatives).

**Comment by Tealk** [comment]
So for me, the language selection is about choosing the language, there is also German in Austria and Switzerland, but for me it's all German.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/LemmyNet/lemmy-ui/issues/1166#issuecomment-2600875602

@Tealk, I don't quite understand what you mean. However, even if those differences don't matter to you, they can to others. Perhaps you'd do better to request a filter option that ignores nation-specific variants of languages, if it bothers you significantly?

**Comment by Tealk** [comment]
But that doesn't solve the problem that you can't filter by language. In the end, the point is that you can only display posts that are written in languages you understand.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/LemmyNet/lemmy-ui/issues/1166#issuecomment-2600879665

@Tealk, in which case, you've fallen victim to the X/Y problem. You've suggested a solution to a problem, instead of requesting that the problem be remediated. Were you to instead request that users be able to see languages outside of those that they have selected, I would agree. Instead, you've requested that the indeterminate option be removed.

**Comment by Tealk** [comment]
Personally, I am more interested in no longer having an undefined language to choose from.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/LemmyNet/lemmy-ui/issues/1166#issuecomment-2600890627

@Tealk, if, for the aforediscussed, my advice remains. Otherwise, you've again neglected to state your ultimate rationale. If you've multiple reasons for requesting this, they should be explicitly stated in the issue description.
